coordinator/src/services/translation/request/dispense/dispense-notification.ts

```diff
@@ -291,15 +313,11 @@ function createSuppliedLineItemQuantity(
   return hl7SuppliedLineItemQuantity
 }
 
-export function getOrganisationPerformer(fhirFirstMedicationDispense: fhir.MedicationDispense): fhir.DispensePerformer {
-  return fhirFirstMedicationDispense.performer.find(p => p.actor.type === "Organization")
-}
-
-export function getMedicationDispenseContained(
-  fhirMedicationDispense: fhir.MedicationDispense
-): fhir.MedicationRequest{
-  const authorizingPrescriptionReference = fhirMedicationDispense.authorizingPrescription[0].reference.replace("#", "")
-  return fhirMedicationDispense.contained.find(p => p.id === authorizingPrescriptionReference)
+export function getMedicationDispenseContained<T extends fhir.MedicationRequest | fhir.PractitionerRole>(
+  fhirMedicationDispense: fhir.MedicationDispense,
+  referenceId: string,
```

Comment:
look at passing type guards into this function to avoid `as T` (for example `isMedicationRequest`)

Reply:
moved to util function using type guard